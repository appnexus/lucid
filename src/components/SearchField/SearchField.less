@import (reference) '../../styles/variables.less';
@import (reference) '../../styles/mixins.less';

.lucid-SearchField {

	position: relative;

	.lucid-TextField {

		border-radius: @TextField-size-height / 2;
		padding: 0 @size-standard;

	}

	&-Icon-container {
		position: absolute;
		top: 0;
		right: 0;
		height: 100%;
		display: flex;
		align-items: center;
		margin-right: @size-XS;
	}

	&-Icon {
		fill: @color-mediumGray;
		&-active {
			fill: @color-primary;
		}
	}

}
