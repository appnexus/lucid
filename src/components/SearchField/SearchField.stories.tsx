import React from 'react';
import createClass from 'create-react-class';
import { LoadingIcon, SearchFieldDumb as SearchField } from './../../index';
import Button from '../Button/Button';

export default {
	title: 'Controls/SearchField',
	component: SearchField,
	parameters: {
		docs: {
			description: {
				component: (SearchField as any).peek.description,
			},
		},
	},
};

/* Default */
export const Default = () => {
	const Component = createClass({
		render() {
			return <SearchField />;
		},
	});

	return <Component />;
};
Default.storyName = 'Default';

/* Interactive */
export const Interactive = () => {
	const Component = createClass({
		getInitialState: () => ({}),
		render() {
			return (
				<div>
					<SearchField onSubmit={(value) => this.setState({ value })} />
					<div style={{ marginTop: '10px', marginLeft: '10px' }}>
						Hit "enter" to submit: {this.state.value}
					</div>
				</div>
			);
		},
	});

	return <Component />;
};
Interactive.storyName = 'Interactive';

/* Placeholder */
export const Placeholder = () => {
	const Component = createClass({
		render() {
			return <SearchField placeholder='Name/ID' />;
		},
	});

	return <Component />;
};
Placeholder.storyName = 'Placeholder';

/* Disabled */
export const Disabled = () => {
	const Component = createClass({
		render() {
			return <SearchField isDisabled />;
		},
	});

	return <Component />;
};
Disabled.storyName = 'Disabled';

/* Custom Icon */
export const CustomIcon = () => {
	const Component = createClass({
		render() {
			return (
				<SearchField>
					<SearchField.Icon>
						<LoadingIcon />
					</SearchField.Icon>
				</SearchField>
			);
		},
	});

	return <Component />;
};
CustomIcon.storyName = 'CustomIcon';

/* Custom Text Field */
export const CustomTextField = () => {
	const Component = createClass({
		getInitialState: () => ({}),
		render() {
			return (
				<div>
					<SearchField placeholder='Name/ID'>
						<SearchField.TextField
							value={this.state.value}
							onSubmit={(submission) => this.setState({ submission })}
							onChange={(value) => this.setState({ value })}
							onKeyDown={({ event: { key } }) => this.setState({ key })}
							onBlur={(lastValue) => this.setState({ lastValue })}
						/>
					</SearchField>
					<div style={{ marginTop: '10px', marginLeft: '10px' }}>
						<div>Hit "enter" to submit: {this.state.submission}</div>
						<div>Last keydown: {this.state.key}</div>
						<div>Value on blur: {this.state.lastValue}</div>
					</div>
				</div>
			);
		},
	});

	return <Component />;
};
CustomTextField.storyName = 'CustomTextField';

/* Valid Search */
export const ValidSearch = () => {
	const Component = createClass({
		getInitialState: () => ({ value: '' }),
		render() {
			return (
				<SearchField
					placeholder="Search icon doesn't become active until you type at least three characters ----->"
					isValid={this.state.value.length > 2}
					value={this.state.value}
					onSubmit={(submission) => this.setState({ submission })}
					onChange={(value) => this.setState({ value })}
					onKeyDown={({ key }) => this.setState({ key })}
					onBlur={(lastValue) => this.setState({ lastValue })}
				/>
			);
		},
	});

	return <Component />;
};
ValidSearch.storyName = 'ValidSearch';

/* Props */
export const Props = () => {
	const Component = createClass({
		render() {
			return (
				<div>
					<SearchField value='foo' />
					<SearchField placeholder='bar' />
					<SearchField isDisabled />
					<SearchField isValid={false} />
					<SearchField>
						<SearchField.Icon>
							<LoadingIcon />
						</SearchField.Icon>
					</SearchField>
					<SearchField value='foo'>
						<SearchField.Icon>
							<LoadingIcon />
						</SearchField.Icon>
					</SearchField>
					<SearchField>
						<SearchField.TextField value='bar' />
					</SearchField>
				</div>
			);
		},
	});

	return <Component />;
};
Props.storyName = 'Props';

/* Debounced */
export const Debounced = () => {
	const Component = createClass({
		getInitialState() {
			return {
				value: 'foo',
			};
		},

		render() {
			return (
				<div>
					<SearchField
						style={{ marginBottom: '10px' }}
						value={this.state.value}
						onChangeDebounced={(value) => this.setState({ value })}
					/>

					<div
						style={{
							marginBottom: '10px',
							marginLeft: '10px',
						}}
					>
						this.state.value: {this.state.value}
					</div>

					<Button
						onClick={() => {
							this.setState({ value: 'foo' });
						}}
					>
						Set TextField to "foo"
					</Button>
				</div>
			);
		},
	});

	return <Component />;
};
Debounced.storyName = 'Debounced';
