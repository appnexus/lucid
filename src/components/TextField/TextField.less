@import (reference) '../../styles/variables.less';
@import (reference) '../../styles/mixins.less';

@TextField-size-padding: @size-S;
@TextField-size-height: @size-standard-height;
@TextField-color-font: #333;

.lucid-TextField {
	.box-sizing();
	.inset-box-shadow();

	background-color: @color-white;
	border-radius: @size-borderRadius;
	border: @border-standardBorder;
	color: @TextField-color-font;
	font-family: @fontFamily;
	font-size: @fontSize;
	margin: 0;
	padding: 0 @TextField-size-padding;
	width: 100%;

	&:focus {
		outline: none;
		.focus-glow();
		border-color: @color-primary;
	}

	&&-is-single-line {
		vertical-align: middle;
		height: @TextField-size-height;
	}

	&&-is-multi-line {
		vertical-align: top;
		resize: both;
		padding-top: @TextField-size-padding - 2px; // Handle padding appropriately for `textarea`s
	}

	&&-is-disabled {
		cursor: not-allowed;
		background-color: @color-gray-5;
		border-color: @color-gray-10;
		color: @color-gray-25;
	}
}
