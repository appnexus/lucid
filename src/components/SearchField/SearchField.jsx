import _ from 'lodash';
import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import {
	createClass as createReactClass,
	getFirst,
	omitProps,
} from '../../util/component-types';
import { buildHybridComponent } from '../../util/state-management';

import TextField from '../TextField/TextField';
import SearchIcon from '../Icon/SearchIcon/SearchIcon';

import reducers from './SearchField.reducers';

const cx = lucidClassNames.bind('&-SearchField');

const { bool, func, node, number, oneOfType, string } = PropTypes;

/**
* {"categories": ["controls", "text"], "madeFrom": ["TextField", "SearchIcon"]}
*
* This is a wrapper around TextField that styles it for a search use-case. The
* icon and TextField are customizable through child components.
*/
const SearchField = createReactClass({
	displayName: 'SearchField',

	components: {
		/**
		 * Allows for full control over the TextField that's used under the hood.
		 */
		TextField,
		/**
		 * Icon this is displayed on the right side of the SearchField. Any of the
		 * lucid `*Icon` components should work.
		 */
		Icon: createReactClass({
			displayName: 'SearchField.Icon',
			propName: 'Icon',
		}),
	},

	reducers,

	propTypes: {
		/**
		 * Fires an event every time the user types text into the TextField.
		 *
		 * Signature: `(value, { event, props }) => {}`
		 */
		onChange: func,
		/**
		 * Fires an event, debounced by `debounceLevel`, when the user types text
		 * into the TextField.
		 *
		 * Signature: `(value, { event, props }) => {}`
		 */
		onChangeDebounced: func,
		/**
		 * Number of milliseconds to debounce the `onChangeDebounced` callback.
		 * Only useful if you provide an `onChangeDebounced` handler.
		 */
		debounceLevel: number,
		/**
		 * Fires an event when the user hits "enter" from the SearchField.
		 *
		 * Signature: `(value, { event, props }) => {}`
		 */
		onSubmit: func,
		/**
		 * Set the value of the input.
		 */
		value: oneOfType([number, string]),
		/**
		 * Controls the highlighting of the search icon. Should be passed `true` when
		 * the search text is valid, e.g. contains enough characters to perform a search.
		 */
		isValid: bool,
		/**
		 * Disables the SearchField by greying it out.
		 */
		isDisabled: bool,
		/**
		 * placeholder value
		 */
		placeholder: string,
		/**
		* Appended to the component-specific class names set on the root
		* element.
		*/
		className: string,
		/**
		 * Icon this is displayed on the right side of the SearchField. Any of the
		 * lucid `*Icon` components should work.
		 */
		Icon: node,
	},

	getDefaultProps() {
		return {
			isDisabled: false,
			onChange: _.noop,
			onChangeDebounced: _.noop,
			debounceLevel: 500,
			onSubmit: _.noop,
			value: '',
		};
	},

	render() {
		const {
			props,
			props: {
				className,
				isDisabled,
				isValid,
				onChange,
				onChangeDebounced,
				debounceLevel,
				onSubmit,
				placeholder,
				value,
				...passThroughs
			},
		} = this;

		const { Icon, TextField } = SearchField;

		const textFieldProps = {
			isDisabled,
			onChange,
			onChangeDebounced,
			debounceLevel,
			onSubmit,
			placeholder,
			isMultiLine: false,
			value,
		};

		const textFieldElement = getFirst(
			props,
			TextField,
			<TextField {...textFieldProps} />
		);
		const isIconActive = _.isUndefined(isValid)
			? !_.isEmpty(_.get(textFieldElement, 'props.value'))
			: isValid;
		const defaultIcon = (
			<SearchIcon className={cx('&-Icon', { '&-Icon-active': isIconActive })} />
		);
		const iconElement = getFirst(props, Icon);
		const iconChildren = _.get(iconElement, 'props.children');
		const icon = iconChildren
			? createElement(iconChildren.type, {
					...iconChildren.props,
					className: cx(
						'&-Icon',
						{ '&-Icon-active': isIconActive },
						iconChildren.props.className
					),
				})
			: defaultIcon;

		return (
			<div
				{...omitProps(passThroughs, SearchField)}
				className={cx('&', className)}
			>
				{textFieldElement}
				<div className={cx('&-Icon-container')}>
					{icon}
				</div>
			</div>
		);
	},
});

export default buildHybridComponent(SearchField);
export { SearchField as SearchFieldDumb };
