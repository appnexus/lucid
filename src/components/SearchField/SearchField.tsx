/* eslint-disable react/prop-types */
import _ from 'lodash';
import React, { createElement } from 'react';
import PropTypes from 'react-peek/prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import {
	StandardProps,
	getFirst,
	omitProps,
	Overwrite,
} from '../../util/component-types';
import { buildModernHybridComponent } from '../../util/state-management';
import TextField, { ITextFieldProps } from '../TextField/TextField';
import SearchIcon from '../Icon/SearchIcon/SearchIcon';
import reducers from './SearchField.reducers';
import { ITextFieldState } from '../TextField/TextField';

const cx = lucidClassNames.bind('&-SearchField');

const { bool, func, node, number, oneOfType, string } = PropTypes;

interface ISearchFieldIcon extends StandardProps {}

const SearchFieldIcon = (_props: ISearchFieldIcon): null => null;
SearchFieldIcon.peek = {
	description: `
		Icon this is displayed on the right side of the SearchField. Any of the lucid \`*Icon\` components should work.
	`,
};
SearchFieldIcon.displayName = 'SearchField.Icon';
SearchFieldIcon.propName = 'Icon';

interface ISearchFieldTextField extends ITextFieldProps {}

const SearchFieldTextField = (_props: ISearchFieldTextField): null => null;
SearchFieldTextField.peek = {
	description: `
		Icon this is displayed on the right side of the SearchField. Any of the lucid \`*Icon\` components should work.
	`,
};
SearchFieldTextField.displayName = 'SearchField.TextField';
SearchFieldTextField.propName = 'TextField';

export interface ISearchFieldState extends ITextFieldState {}

export interface ISearchFieldProps extends StandardProps {
	/** Fires an event every time the user types text into the TextField. */
	onChange: (
		value: string,
		{ event, props }: { event: React.FormEvent; props: ITextFieldProps }
	) => void;

	/** Fires an event, debounced by \`debounceLevel\`, when the user types text
	into the TextField. */
	onChangeDebounced?: (
		value: string,
		{ event, props }: { event: React.FormEvent; props: ITextFieldProps }
	) => void;

	/** Number of milliseconds to debounce the \`onChangeDebounced\` callback.
			Only useful if you provide an \`onChangeDebounced\` handler. */
	debounceLevel: number;

	/** Fires an event when the user hits "enter" from the SearchField. */
	onSubmit: (
		value: string,
		{ event, props }: { event: React.FormEvent; props: ITextFieldProps }
	) => void;

	/** Set the value of the input. */
	value: string | number;

	/** Controls the highlighting of the search icon. Should be passed \`true\`
			when the search text is valid, e.g. contains enough characters to perform
			a search. */
	isValid?: boolean;

	/** Disables the SearchField by greying it out. */
	isDisabled: boolean;

	/**n placeholder value */
	placeholder?: string;
}

type ISearchFieldPropsWithPassThroughs = Overwrite<
	React.DetailedHTMLProps<
		React.InputHTMLAttributes<HTMLInputElement>,
		HTMLInputElement
	>,
	ISearchFieldProps
>;

class SearchField extends React.Component<
	ISearchFieldPropsWithPassThroughs,
	ISearchFieldState
> {
	static displayName = 'SearchField';
	static TextField = SearchFieldTextField;
	static Icon = SearchFieldIcon;

	static peek = {
		description: `
					This is a wrapper around TextField that styles it for a search
					use-case. The icon and TextField are customizable through child
					components.
				`,
		categories: ['controls', 'text'],
		madeFrom: ['TextField', 'SearchIcon'],
	};

	static reducers = reducers;

	static propTypes = {
		onChange: func`
			Fires an event every time the user types text into the TextField.
			Signature: \`(value, { event, props }) => {}\`
		`,

		onChangeDebounced: func`
			Fires an event, debounced by \`debounceLevel\`, when the user types text
			into the TextField.  Signature: \`(value, { event, props }) => {}\`
		`,

		debounceLevel: number`
			Number of milliseconds to debounce the \`onChangeDebounced\` callback.
			Only useful if you provide an \`onChangeDebounced\` handler.
		`,

		onSubmit: func`
			Fires an event when the user hits "enter" from the SearchField.
			Signature: \`(value, { event, props }) => {}\`
		`,

		value: oneOfType([number, string])`
			Set the value of the input.
		`,

		isValid: bool`
			Controls the highlighting of the search icon. Should be passed \`true\`
			when the search text is valid, e.g. contains enough characters to perform
			a search.
		`,

		isDisabled: bool`
			Disables the SearchField by greying it out.
		`,

		placeholder: string`
			placeholder value
		`,

		className: string`
			Appended to the component-specific class names set on the root element.
		`,

		Icon: node`
			Icon this is displayed on the right side of the SearchField. Any of the
			lucid \`*Icon\` components should work.
		`,

		TextField: node`
			The TextField that Searchfield is composed of.
		`,
	};

	static defaultProps = {
		isDisabled: false,
		onChange: _.noop,
		onChangeDebounced: _.noop,
		debounceLevel: 500,
		onSubmit: _.noop,
		value: '',
	};

	render(): React.ReactNode {
		const {
			props,
			props: {
				className,
				isDisabled,
				isValid,
				onChange,
				onChangeDebounced,
				debounceLevel,
				onSubmit,
				placeholder,
				value,
				autoComplete,
				...passThroughs
			},
		} = this;

		const { Icon } = SearchField;

		const textFieldProps = _.get(
			getFirst(props, SearchField.TextField),
			'props'
		) || {
			isDisabled,
			onChange,
			onChangeDebounced,
			debounceLevel,
			onSubmit,
			placeholder,
			isMultiLine: false,
			value,
			autoComplete,
		};

		const textFieldElement = <TextField {...textFieldProps} />;
		const isIconActive = _.isUndefined(isValid)
			? !_.isEmpty(_.get(textFieldElement, 'props.value'))
			: isValid;
		const defaultIcon = (
			<SearchIcon
				size={12}
				className={cx('&-Icon', { '&-Icon-active': isIconActive })}
			/>
		);
		const iconElement = getFirst(props, Icon);
		const iconChildren = _.get(iconElement, 'props.children');
		const icon = iconChildren
			? createElement(iconChildren.type, {
					...iconChildren.props,
					className: cx(
						'&-Icon',
						{ '&-Icon-active': isIconActive },
						iconChildren.props.className
					),
			  })
			: defaultIcon;

		return (
			<div
				{...omitProps(passThroughs, undefined, _.keys(SearchField.propTypes))}
				className={cx('&', className)}
			>
				{textFieldElement}
				<div className={cx('&-Icon-container')}>{icon}</div>
			</div>
		);
	}
}

export default buildModernHybridComponent<
	ISearchFieldPropsWithPassThroughs,
	ISearchFieldState,
	typeof SearchField
>(SearchField, { reducers });
export { SearchField as SearchFieldDumb };
