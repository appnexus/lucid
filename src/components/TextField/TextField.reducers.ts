import { ITextFieldState } from './TextField';

export function onChange(
	state: ITextFieldState,
	value: number | string
): ITextFieldState {
	return {
		...state,
		value: value,
	};
}

export default { onChange };
