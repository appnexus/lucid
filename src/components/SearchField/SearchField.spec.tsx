import React from 'react';
import { shallow } from 'enzyme';
import assert from 'assert';
import { common } from '../../util/generic-tests';
import { SearchFieldDumb as SearchField } from './SearchField';
import TextField from '../TextField/TextField';

describe('SearchField', () => {
	common(SearchField);

	describe('props', () => {
		const fn = () => {};

		describe('onChange', () => {
			it('should pass onChange to the text input', () => {
				const wrapper = shallow(<SearchField onChange={fn} />);
				const textFieldWrapper = wrapper.find(TextField).first();
				assert.equal(textFieldWrapper.props().onChange, fn);
			});
		});

		describe('onChangeDebounced', () => {
			it('should pass onChangeDebounced to the text input', () => {
				const wrapper = shallow(<SearchField onChangeDebounced={fn} />);
				const textFieldWrapper = wrapper.find(TextField).first();
				assert.equal(textFieldWrapper.props().onChangeDebounced, fn);
			});
		});

		describe('onSubmit', () => {
			it('should pass onSubmit to the text input', () => {
				const wrapper = shallow(<SearchField onSubmit={fn} />);
				const textFieldWrapper = wrapper.find(TextField).first();
				assert.equal(textFieldWrapper.props().onSubmit, fn);
			});
		});
	});
});
