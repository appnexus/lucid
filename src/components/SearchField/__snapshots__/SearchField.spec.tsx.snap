// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField [common] example testing should match snapshot(s) for 1.default 1`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 2.interactive 1`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 3.placeholder 1`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    placeholder="Name/ID"
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 4.disabled 1`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={true}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 5.custom-icon 1`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <LoadingIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={16}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 6.custom-text-field 1`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 7.valid-search 1`] = `
<div
  className="lucid-SearchField"
  onBlur={[Function]}
  onKeyDown={[Function]}
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    placeholder="icon doesn't become active until you type at least three characters ----->"
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 8.props 1`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value="foo"
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon lucid-SearchField-Icon-active"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 8.props 2`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    placeholder="bar"
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 8.props 3`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={true}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 8.props 4`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 8.props 5`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value=""
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <LoadingIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={16}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 8.props 6`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value="foo"
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <LoadingIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon lucid-SearchField-Icon-active"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={16}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 8.props 7`] = `
<div
  className="lucid-SearchField"
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value="bar"
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon lucid-SearchField-Icon-active"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;

exports[`SearchField [common] example testing should match snapshot(s) for 9.debounced 1`] = `
<div
  className="lucid-SearchField"
  style={
    Object {
      "marginBottom": "10px",
    }
  }
>
  <TextField
    debounceLevel={500}
    isDisabled={false}
    isMultiLine={false}
    lazyLevel={1000}
    onBlur={[Function]}
    onChange={[Function]}
    onChangeDebounced={[Function]}
    onKeyDown={[Function]}
    onSubmit={[Function]}
    rows={5}
    value="foo"
  />
  <div
    className="lucid-SearchField-Icon-container"
  >
    <SearchIcon
      aspectRatio="xMidYMid meet"
      className="lucid-SearchField-Icon lucid-SearchField-Icon-active"
      color="primary"
      isClickable={false}
      isDisabled={false}
      onClick={[Function]}
      onSelect={[Function]}
      size={12}
      viewBox="0 0 16 16"
    />
  </div>
</div>
`;
